File: ruby/helloweb.rb
# This code requires the sinatra gem to be installed.
# Run the application by typing:
#
# ruby helloweb.rb
#
# Go to:
# http://localhost:4567/Hello/stranger
# http://localhost:4567/Hi/stranger

require 'sinatra'

get '/hello/:name' do |name|
	"Hello #{name}!"
end

get('/hi/:name') { |name| "Hi #{name}" }

# To see how get is implemented please look at sinatra 
# source code.
#
# http://github.com/sinatra/sinatra

File: ruby/helloweb.cs
namespace WYSTTS.Ruby
{
	using System;
	using System.Dynamic;

	public class Example
	{
		public void LookAtMe()
		{
			Get("/hello/:name", (param) => {
				return String.Format("Hello {0}!", param.name);
			});
		}

		private void Get(string path, Func<dynamic, string> param)
		{
			// Do your cool stuff here :)
		}

		/*

		Other usages of this where demonstrated using NancyFx.
		To find Nancy samples, documentation and source code
		please checkout:

		http://github.com/NancyFx/Nancy
		
		*/
	}

	public static class Program
	{
		public static void Main(string[] args) {}
	}
}